test/lib/Resource.js

```diff
@@ -316,6 +316,31 @@ test("Resource: clone resource with stream", async (t) => {
 	t.is(clonedResourceContent, "Content", "Cloned resource has correct content string");
 });
 
+test("Resource: clone resource with project", async (t) => {
+	t.plan(3);
+
+	const myProject = {
+		name: "my project"
+	};
+	const resourceOptions = {
+		path: "my/path/to/resource",
+		project: myProject
+	};
+
+	const resource = new Resource({
+		path: "my/path/to/resource",
+		project: myProject
+	});
+
+	const clonedResource = await resource.clone();
+	t.pass("Resource cloned");
+
+	const clonedResourceProject = await clonedResource.getProject();
+	t.deepEqual(clonedResourceProject, resourceOptions.project, "Cloned resource has correct project content");
```

Comment:
I would think that the t.is assert in the next line is sufficient. If the two variables hold the same instance, they are naturally deeply-equal too, are they not?

Reply:
Thats true... in one of the commits before i have cloned the project too, so the instance check was false but the deepequal check was true. Will remove this assert